go/vt/servenv/grpc_server.go

```diff
@@ -145,10 +147,19 @@ func createGRPCServer() {
 		opts = append(opts, grpc.UnaryInterceptor(unaryInterceptor))
 	}
 
+	if *grpccommon.EnableGRPCPrometheus {
+		opts = append(opts, grpc.StreamInterceptor(grpc_prometheus.StreamServerInterceptor))
+		opts = append(opts, grpc.UnaryInterceptor(grpc_prometheus.UnaryServerInterceptor))
+	}
+
 	GRPCServer = grpc.NewServer(opts...)
 }
 
 func serveGRPC() {
+	if *grpccommon.EnableGRPCPrometheus {
+		grpc_prometheus.Register(GRPCServer)
+		grpc_prometheus.EnableHandlingTimeHistogram()
```

Comment:
From my read of the docs this is disabled by default because of a concern about high cardinality:
https://github.com/grpc-ecosystem/go-grpc-prometheus#histograms

However it _looks_ like there will be a tag for each service/method pair, which (in v3 at least) is actually pretty tight, so it seems like this is ok?

Still it would be good if you would confirm this on vtgate and vttablet so that we know in practice what this will do to our cardinality.

Reply:
Yep! Exactly:

From the gates, an e.g.:

```
grpc_server_handling_seconds_bucket{grpc_method="ExecuteFetchAsDba",grpc_service="tabletmanagerservice.TabletManager",grpc_type="unary",le="0.005"} 0
```

So, the cardinality is roughly one metric for every rpc _method_ defined, as each method has a unique service label. Taking a look at the services defined in the .proto files, `tabletmanagerservice` has the most at 43 RPC methods, everything else is much less than that. This is within the cardinality that's acceptable for us (Slack), but we could flag this separately if we think it might not be ok for other people's setups. 

IMO, I think it should be fine as Vitess has a well-bounded number of methods / services **and** there are *so few* metrics output by the package generally that the histogram high-cardinality metric-explosion is relatively OK.